### DISCUSSION TITLE
rss-parser

### DISCUSSION THREAD

**Discussion by craigz** [johnlindquist/kit#696] [Q&A] [answered]
I've got a node script that uses an npm library `rss-parse` and simply outputs the title and url for each story in the feed to console.log (and i've also used the 'fs' module to write the resulting .json to a file along the way in testing). 

The script is working just fine, running via node/cli. 

To bring it into scriptkit I started by duplicating the hackernews example.
However that script at some point uses the map function to create a new array from the response. Having changed the variable names to those in my feed generates an error when i run the script. 

`TypeError: Cannot read properties of undefined (reading 'map')`

I was able to have scriptkit install the npm module for me by the way, so i know that isn't the issue at the moment.

I was first trying to use `inspect(response.data)` to see exactly what was coming in and being rejected, however with that line in the code uncommented, the window simply goes away. I've spent some time plundering the various log, temp, etc folders in `~/.kenv/*` and `~/.kit/*`  to locate the dumped .json file, but can't find it anywhere to inspect it. 

I've not been able to find any concrete example of how to use the inspect function, so i changed gears and came here. 
Below is the entirety of the script (in scriptkit) that i'm trying to get working, any pointers or links to documentation would be appreciated.  

By the way, I'd been working with a more 'standard' response value initially, however as the script was failing, I ended up pasting in exactly the response value that's working in my `node.js` app.

Here's the content of the script:

```javascript
// Name: daring fireball for scriptkit
// Description: view daring fireball on the quick

import "@johnlindquist/kit";
// import npm library 'rss-parser'
let Parser = await npm("rss-parser");
let parser = new Parser();
let response = await parser.parseURL("https://daringfireball.net/feeds/main");
// inspect(response.data); # dumps all data to json (file). useful to check response data.
let url = await arg(
  `Select Story:`,
  response.data.map(({ title, link }) => {
    return {
      name: title,
      description: link,
      value: link,
      preview: async () => {
        let response = await get(`${link}`);

        return md(`# ${title}`);
      },
    };
  })
);

await $`open ${url}`;
```

**Comment by johnlindquist** [comment, marked as answer]
Looks like the "rss-parser" lib returns a "feed" with "items" on it:


[Open daring-fireball in Script Kit](https://scriptkit.com/api/new?name=daring-fireball&url=https://gist.githubusercontent.com/johnlindquist/c644afc4d1c0cb30a52cdb8abe0d5505/raw/aed62f7f2032515a2ebacdcb7d52867d2c8fb2ea/daring-fireball.ts")

```js
// Name: daring fireball for scriptkit
// Description: view daring fireball on the quick

import "@johnlindquist/kit";

let Parser = await npm("rss-parser");

let parser = new Parser();
let feed = await parser.parseURL("https://daringfireball.net/feeds/main");

let url = await arg(
  `Select Story:`,
  feed.items.map(({ title, link }) => {
    return {
      name: title,
      description: link,
      value: link,
      preview: async () => {
        let response = await get(`${link}`);

        return md(`# ${title}`);
      },
    };
  })
);

await $`open ${url}`;
```

**Reply by craigz** [reply]
indeed, that did the trick 🎉 
thanks so much! 

I'd been ripping my hair out on that one since sometime last night.

**Reply by johnlindquist** [reply]
Always happy to help!
